core/src/main/java/org/apache/iceberg/hadoop/HadoopStreams.java

```diff
@@ -185,8 +185,20 @@ public void flush() throws IOException {
 
     @Override
     public void close() throws IOException {
-      stream.close();
-      this.closed = true;
+      try {
+        stream.close();
+        // {@link org.apache.hadoop.fs.s3a.S3ABlockOutputStream#close()} calls {@link
+        // org.apache.hadoop.fs.s3a.S3ABlockOutputStream#putObject()}
+        // which doesn't throw an exception when interrupted. Need to handle interrupts here to make
```

Comment:
nit: `make sure` is not accurate here. It is more like `Need to check the interrupted flag to detect failed object upload and propagate the error up`.

Reply:
Done.